packages/scaffold-ui/src/partials/w3m-all-wallets-list-item/index.ts

```diff
@@ -0,0 +1,101 @@
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import { property, state } from 'lit/decorators.js'
+import { ifDefined } from 'lit/directives/if-defined.js'
+import { AssetUtil, type WcWallet } from '@web3modal/core'
+import styles from './styles.js'
+
+@customElement('w3m-all-wallets-list-item')
+export class W3mAllWalletsListItem extends LitElement {
+  public static override styles = styles
+
+  // -- Members ------------------------------------------- //
+  private observer = new IntersectionObserver(() => undefined)
+
+  // -- State & Properties -------------------------------- //
+  @state() private visible = false
+
+  @state() private imageSrc: string | undefined = undefined
+
+  @state() private imageLoading = false
+
+  @property() private wallet: (WcWallet & { installed: boolean }) | undefined = undefined
+
+  // -- Lifecycle ----------------------------------------- //
+  constructor() {
+    super()
+    this.observer = new IntersectionObserver(
+      entries => {
+        entries.forEach(entry => {
+          if (entry.isIntersecting) {
+            this.visible = true
+            this.fetchImageSrc()
```

Comment:
Is there any way we could request this data concurrently? 
I mean we have like 4 wallets per line, would be 4 requests, could we request those 4 concurrently? wyt?

Reply:
I think we need to move the intersectionObserve to the parent component `w3m-all-wallets-list`. We can add the id of the wallet as a data-attribute, so we can store this data and fetch 4 at the same time?